packages/oauth/oauth-client/src/oauth-authorization-server-metadata-resolver.ts

```diff
@@ -31,18 +35,30 @@ export class OAuthAuthorizationServerMetadataResolver extends CachedGetter<
   OAuthAuthorizationServerMetadata
 > {
   private readonly fetch: Fetch<unknown>
+  private readonly allowUnsecure: boolean
 
-  constructor(cache: AuthorizationServerMetadataCache, fetch?: Fetch) {
+  constructor(
+    cache: AuthorizationServerMetadataCache,
+    fetch?: Fetch,
+    config?: OAuthAuthorizationServerMetadataResolverConfig,
+  ) {
     super(async (issuer, options) => this.fetchMetadata(issuer, options), cache)
 
     this.fetch = bindFetch(fetch)
+    this.allowUnsecure = config?.allowUnsecure === true
   }
 
   async get(
-    issuer: string,
+    input: string,
     options?: GetCachedOptions,
   ): Promise<OAuthAuthorizationServerMetadata> {
-    return super.get(oauthIssuerIdentifierSchema.parse(issuer), options)
+    const issuer = oauthIssuerIdentifierSchema.parse(input)
+    if (!this.allowUnsecure && !issuer.startsWith('https:')) {
```

Comment:
Small nit, but I think we want to use "insecure" rather than "unsecure."  We could also make it more specific, referencing whether "http" is allowed.